BraveWallet/Crypto/BuySwapSwap/SendTokenView.swift

```diff
@@ -82,11 +82,11 @@ struct SendTokenView: View {
           )
             .onChange(of: amountInput, perform: { value in
               guard let token = sendTokenStore.selectedSendToken else { return }
-              // will ignore if input is not a decimal number
-              if let decimal = Decimal(string: value), -decimal.exponent > 0 {
-                if -decimal.exponent > Int(token.decimals) {
-                  self.amountInput = String(value.prefix(Int(token.decimals) + 2))
-                }
+              let weiFormatter = WeiFormatter(decimalFormatStyle: .decimals(precision: Int(token.decimals)))
+              
+              guard weiFormatter.weiString(from: value, radix: .decimal, decimals: Int(token.decimals)) != nil else {
+                self.amountInput = String(value.dropLast())
+                return
               }
```

Comment:
1. Given this is the end of the scope, you may as well just use `if` without a return instead of `guard-else`.
2. If a user pastes a really large number with too many decimals, won't this execute over and over until it eventually passes? What if we instead just disable the button for invalid amounts rather than try and correct them?